[WIP] RHDEVDOCS-3819: [UG] How to store and retrieve logs (#60966)
openshift/openshift-docs: modules/gitops-storing-and-retrieving-argo-cd-logs.adoc

// Module included in the following assemblies:
//
// * cicd/gitops/viewing-argo-cd-logs-using-the-kibana-dashboard.adoc

:_content-type: PROCEDURE
[id="gitops-storing-and-retrieving-argo-cd-logs_{context}"]
= Storing and retrieving Argo CD logs

> You can use the Kibana dashboard to store and retrieve Argo CD logs.

Reviewer: An assembly with one module included is OK. Maybe in the future, other procedures will be added to this assembly file. We don't know what we don't know, but this possibility is always there.

.Procedure

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Author: Hello @varshab1210 I had the same issue as you while testing it, that is why in my PR I added as a prerequisite to install the whole logging subsystem and not just Red Hat OpenShift Logging Operator so I am not sure what exactly should I change. Can you please specify? Thank you!

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Reviewer 2: Hey @ekristova I have followed the section of "Installing the logging subsystem for Red Hat OpenShift using the web console" from https://docs.openshift.com/container-platform/4.11/logging/cluster-logging-deploying.html and it worked for me. I am able to see `Logging` option, like this:
![Screenshot from 2023-06-29 17-24-29](https://github.com/openshift/openshift-docs/assets/73488906/79c7d537-fc25-46df-9d49-aafc14749d01)

And when I am visiting it, I am getting:

![Screenshot from 2023-06-29 17-25-55](https://github.com/openshift/openshift-docs/assets/73488906/2bfb4e65-a6a6-48d3-9668-7ff89e7b93b0)

Maybe it takes time to get up the dashboard?
If it is so, you can mention the approx timing in the first step.

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Author: Hello @iam-veeramalla do you know why this problem might be occurring?

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Reviewer 2: I see now that pods are having some error, see all the pods under `openshift-logging` project:

![Screenshot from 2023-06-30 12-35-53](https://github.com/openshift/openshift-docs/assets/73488906/a7520ef7-e009-4d5b-995e-6820b886862f)

And now when I see inside the errored pods that is `elasticsearch-im-app-28135140-wwbhb` ,  `elasticsearch-im-audit-28135140-x4nvc`, and `elasticsearch-im-infra-28135140-cpc6k`, I got (for all three of them, same error):

![Screenshot from 2023-06-30 12-42-40](https://github.com/openshift/openshift-docs/assets/73488906/5b9a888a-5bb2-4009-bad3-6e3c7bfacecf)

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Author: Hello @iam-veeramalla as seen in the above thread, there seems to be a problem with a prerequisite. Is there something more that needs to be done, or is there some problem with the prerequisite itself? If that is the case then  what would be the best way to proceed (as that would be out of scope)?

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Reviewer 3: Hi @ekristova and @mehabhalodiya , the above error state that the cluster you are using lacks resources. Please use a cluster with nodes(resources) available for scheduling the pods.

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Reviewer 4: Hi @iam-veeramalla, the above used cluster had 3 worker and master nodes. Can you suggest some additional configuration so that this can be avoided?

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Reviewer 3: That configuration is fine @varshab1210 but if you look at the error, the resources may be consumed by some of other workloads running on the cluster.

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Reviewer 4: I was able to reproduce the steps 

![Screenshot from 2023-08-08 14-51-13](https://github.com/openshift/openshift-docs/assets/60093147/c3aa2c9b-8488-41c2-8627-bb240d69f43c)

> . In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. The Kibana dashboard is opened in a new window.

Reviewer 5: You can get rid of the passive voice here:
```suggestion
. In the {product-title} web console, navigate to the {rh-app-icon} menu -> *Observability* -> *Logging*. This opens the Kibana dashboard in a new window.
```

Reviewer: +1

. Create an index pattern.

> .. To display all the indeces, define the index pattern as "***" . Click *Next step* to continue.

Reviewer 5: Nitpick: in my opinion, there is no need to say that the purpose of clicking `*Next step*` is `to continue` because this UI label seems self-explanatory to me as a reader.
```suggestion
.. To display all the indeces, define the index pattern as "***" and click *Next step*.
```

> .. To display all the indeces, define the index pattern as "***" . Click *Next step* to continue.

Reviewer: ```suggestion
.. To display all the indices, define the index pattern as `***` . Click *Next step* to continue.
```
- fix typo
- Enclose in back ticks the literal text to be entered by the user, unless the user is also supposed to enter the quotation marks as well.

> .. Select *@timestamp* for _Time Filter field name_.

Reviewer: ```suggestion
.. Select *@timestamp* for *Time Filter field name*.
```
- Whatever the name of the label is in the UI should be in bold type.

> .. Click *Create index pattern* and wait until the index pattern is created.

Reviewer: ```suggestion
.. Click *Create index pattern*.
```
- Consider cutting out the second part of the sentence. It's self-explanatory that clicking **Create index pattern** will create an index pattern.

> . In the left navigation panel of the Kibana dashboard, click the *Discover* tab.

Reviewer: @max-cx is correct. Per the [ISG's accessibility guidelines](https://www.ibm.com/docs/en/ibm-style?topic=medium-accessibility), don't use "sensory words such as Right, Left, Square, Triangle, Round, Above, or Below to describe elements in the UI, or a graphical symbol alone to convey information. Some users cannot perceive elements that are a certain picture, color, shape or in a particular location."

> . Create a filter to retrieve logs for Argo CD.

Reviewer: ```suggestion
. Create a filter to retrieve logs for Argo CD. The following steps create a filter that retrieves logs for all the pods in the `openshift-gitops` namespace:
```

.. Click *Add a filter +*.

> .. Provide `kubernetes.namespace_name` as the field. Select the "*is*" operator and enter `openshift-gitops` as a value.

Reviewer 5: Seems like this list items contains three different steps.

Author: The place where the user enters `kubernetes.namespace_name` is specifically named "Fields", that is why I am calling it "field". However, I will rephrase it as you suggested for better clarity, thanks!

Reviewer: - These are dropdown lists and not user-entered text, so use "select" rather than "provide". 
- Also UI elements, including list items in dropdown lists, should be in bold type.

- Maybe consider creating 3 short steps instead of a single step:

```
.. Select the *kubernetes.namespace_name* namespace. 
.. Select the *is* operator.
.. Select the *openshift-gitops* group.
```

- Also, even though the **Filter** dropdown lists do not have labels, you should add descriptive words as in my example. (I don't know if "namespace" and "group" are the correct names, btw, so you should double-check that.)

+

> This filter retrieves logs for all the pods in the `openshift-gitops` namespace.

Reviewer 5: I recommend moving and incorporating this sentence into the sentence of the step `.  Create a filter to retrieve logs for Argo CD.`. For example:

.  Create a filter to retrieve logs for Argo CD to retrieve logs for all the pods in the `openshift-gitops` namespace.

.. Click *Save* to create the filter.

> . You can add additional filters to narrow the search. For example, to retrieve logs for a particular pod, create another filter with `kubernetes.pod_name` as the field.

Reviewer 5: Nitpick: I'd recommending adding `can` to the example to make it easier to read.
```suggestion
. You can add additional filters to narrow the search. For example, to retrieve logs for a particular pod, you can create another filter with `kubernetes.pod_name` as the field.
```
